### PR TITLE
add Dockerfile

### PR DESCRIPTION
**Pull request by AkihiroSuda**
## Description

add Dockerfile

## Motivation and Context

- [X] I have raised an issue to propose this change **this is required**

https://github.com/openfaas/faasd/issues/42

## How Has This Been Tested?

```console
$ docker build -t faasd . && docker run -td --name faasd -p 127.0.0.1:8080:8080 --privileged faasd
$ docker exec faasd cat /var/lib/faasd/secrets/basic-auth-password | faas-cli login -s
$ faas-cli store deploy figlet
$ echo hello | faas-cli invoke figlet
```

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

Commits:

- [X] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [X] My commit message has a body and describe how this was tested and why it is required.
- [X] I have signed-off my commits with `git commit -s` for the Developer Certificate of Origin (DCO)

Code:

- [X] My code follows the code style of this project.
- [X] I have added tests to cover my changes.

Docs:

- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,7 @@ script:
 - make dist
 - make prepare-test
 - make test-e2e
+- docker build -t dummy-image-for-testing-dockerfile .
```
Comment (alexellis) [inline .travis.yml]: One question - is docker-ce still present as a package in .travis.yml?

Comment (alexellis) [reply .travis.yml]: I don't believe it is, this would fail CI - https://github.com/openfaas/faasd/blob/master/.travis.yml#L10

Comment (AkihiroSuda) [reply .travis.yml]: passing https://travis-ci.com/openfaas/faasd/builds/148533376

Comment (AkihiroSuda) [reply .travis.yml]: What needs to be done?

**Comment by AkihiroSuda** [comment]
ping @alexellis

**Comment by AkihiroSuda** [comment]
ping

**Comment by alexellis** [comment]
@LucasRoesler is this of use to you for testing/dev on Mac?

File: .travis.yml (hunk 1 above)
Comment (alexellis) [reply .travis.yml]: I thought adding `docker-ce` to packages? https://github.com/openfaas/faas/blob/master/.travis.yml#L13

Comment (AkihiroSuda) [reply .travis.yml]: Why? CI seems passing

**Comment by carlosedp** [comment]
LGTM. Works perfectly on Mac.

```
❯ docker build -t faasd . --load && docker run -td --name faasd -p 127.0.0.1:8080:8080 --privileged faasd
[+] Building 28.5s (25/25) FINISHED
 => [internal] load build definition from Dockerfile                                                                                                                                         0.0s
 => => transferring dockerfile: 32B                                                                                                                                                          0.0s
 => [internal] load .dockerignore                                                                                                                                                            0.0s
 => => transferring context: 2B                                                                                                                                                              0.0s
 => [internal] load metadata for docker.io/library/debian:10                                                                                                                                 2.6s
 => [internal] load metadata for docker.io/library/golang:1.13-buster                                                                                                                        1.8s
 => [internal] load build context                                                                                                                                                            0.3s
 => => transferring context: 204.47kB                                                                                                                                                        0.3s
 => [faasd 1/4] FROM docker.io/library/golang:1.13-buster@sha256:1da34145a757a5b9959111d0150d257c1a1f456b5fc3ec72101093e01977d8ca                                                            0.0s
 => => resolve docker.io/library/golang:1.13-buster@sha256:1da34145a757a5b9959111d0150d257c1a1f456b5fc3ec72101093e01977d8ca                                                                  0.0s
 => https://raw.githubusercontent.com/containerd/containerd/v1.3.3/containerd.service                                                                                                        0.1s
 => https://raw.githubusercontent.com/AkihiroSuda/containerized-systemd/b9f98826563f12b36ab5474f172b5082f17d420c/docker-entrypoint.sh                                                        0.1s
 => [stage-1 1/12] FROM docker.io/library/debian:10@sha256:a63d0b2ecbd723da612abf0a8bdb594ee78f18f691d7dc652ac305a490c9b71a                                                                  0.0s
 => => resolve docker.io/library/debian:10@sha256:a63d0b2ecbd723da612abf0a8bdb594ee78f18f691d7dc652ac305a490c9b71a                                                                           0.0s
 => CACHED [stage-1 2/12] RUN apt-get update &&   apt-get install -y --no-install-recommends   ca-certificates   curl   iptables   runc   systemd   systemd-sysv                             0.0s
 => CACHED [stage-1 3/12] RUN mkdir -p /opt/cni/bin &&   curl -sLSf https://github.com/containerd/containerd/releases/download/v1.3.3/containerd-1.3.3.linux-amd64.tar.gz   | tar xzvC /usr  0.0s
 => CACHED [stage-1 4/12] ADD https://raw.githubusercontent.com/containerd/containerd/v1.3.3/containerd.service /etc/systemd/system/                                                         0.0s
 => CACHED [stage-1 5/12] RUN curl -sLSf https://github.com/containernetworking/plugins/releases/download/v0.8.5/cni-plugins-linux-amd64-v0.8.5.tgz   | tar xzvC /opt/cni/bin                0.0s
 => CACHED [stage-1 6/12] RUN curl -sSLf https://cli.openfaas.com | sh                                                                                                                       0.0s
 => CACHED [faasd 2/4] ADD . /go/src/github.com/openfaas/faasd                                                                                                                               0.0s
 => CACHED [faasd 3/4] WORKDIR  /go/src/github.com/openfaas/faasd                                                                                                                            0.0s
 => CACHED [faasd 4/4] RUN make                                                                                                                                                              0.0s
 => CACHED [stage-1 7/12] COPY --from=faasd /go/src/github.com/openfaas/faasd/bin/faasd /usr/local/bin/                                                                                      0.0s
 => CACHED [stage-1 8/12] COPY --from=faasd /go/src/github.com/openfaas/faasd /go/src/github.com/openfaas/faasd                                                                              0.0s
 => CACHED [stage-1 9/12] WORKDIR /go/src/github.com/openfaas/faasd                                                                                                                          0.0s
 => CACHED [stage-1 10/12] ADD https://raw.githubusercontent.com/AkihiroSuda/containerized-systemd/b9f98826563f12b36ab5474f172b5082f17d420c/docker-entrypoint.sh /                           0.0s
 => CACHED [stage-1 11/12] RUN chmod +x /docker-entrypoint.sh                                                                                                                                0.0s
 => CACHED [stage-1 12/12] ADD docker-2ndboot.sh /                                                                                                                                           0.0s
 => exporting to oci image format                                                                                                                                                           20.7s
 => => exporting layers                                                                                                                                                                     11.4s
 => => exporting manifest sha256:2f362a353302494bfac0ead371183736ce5843a2ab38000102bd188546093c65                                                                                            0.0s
 => => exporting config sha256:1ba3b35cd4f6654a0051c83a578f5503f2aeeba7e5bce930622b63e24878ed6b                                                                                              0.0s
 => => sending tarball                                                                                                                                                                       9.1s
 => importing to docker                                                                                                                                                                      5.0s
a153ac594981270b91d2d0f0cd1f7c6eb9664836524d928f3fb1f735aa0b94b5

❯ docker exec faasd cat /var/lib/faasd/secrets/basic-auth-password | faas-cli login -s
Calling the OpenFaaS server to validate the credentials...
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.
credentials saved for admin http://127.0.0.1:8080
❯ faas-cli ls
Function                        Invocations     Replicas
❯ faas-cli store deploy figlet
WARNING! Communication is not secure, please consider using HTTPS. Letsencrypt.org offers free SSL/TLS certificates.

Deployed. 200 OK.
URL: http://127.0.0.1:8080/function/figlet

❯ curl -d test http://127.0.0.1:8080/function/figlet
 _            _
| |_ ___  ___| |_
| __/ _ \/ __| __|
| ||  __/\__ \ |_
 \__\___||___/\__|

❯ uname -a
Darwin Naboo.local 18.7.0 Darwin Kernel Version 18.7.0: Tue Aug 20 16:57:14 PDT 2019; root:xnu-4903.271.2~2/RELEASE_X86_64 x86_64 i386 MacBookPro15,1 Darwin
```

**Comment by LucasRoesler** [comment]
looks good and works for me too

**Comment by Strum355** [comment]
Is there a way to disable basic auth with this?

**Comment by AkihiroSuda** [comment]
ping @alexellis

**Comment by carlosedp** [comment]
Canceled this @AkihiroSuda ?

**Comment by alexellis** [comment]
@LucasRoesler

**Pull request closed without merging**